browser/BUILD.gn

```diff
@@ -190,8 +196,28 @@ source_set("browser_process") {
       "android/preferences/brave_pref_service_bridge.cc",
       "android/preferences/website/desktop_mode_tab_helper.cc",
       "android/preferences/website/desktop_mode_tab_helper.h",
+      "android/rewards/safetynet_check.cc",
+      "android/rewards/safetynet_check.h",
     ]
 
+    if (brave_rewards_enabled) {
+      sources += [
+        "android/rewards/brave_rewards_native_worker.cc",
+        "android/rewards/brave_rewards_native_worker.h",
+      ]
+
+      deps += [
+        "//brave/vendor/bat-native-ledger",
```

Comment:
we shouldn't have a direct dep on bat-native-ledger here and you are missing a dep for`//brave/components/brave_rewards/browser` which already has a dep on bat-native-ledger. You should add deps for every include in your file

Reply:
I guess we have to have both in that case ../../brave/components/brave_rewards/browser/rewards_service_private_observer.h:10:10: fatal error: 'bat/ledger/mojom_structs.h' file not found #include "bat/ledger/mojom_structs.h"

Reply:
I'll take a look because that shouldn't be the case and would indicate another issue

Reply:
oh, actually this is correct and should be here since you have an include for it. Sorry :)